### PR TITLE
clusterversion controller: crash on unparseable poll interval

### PR DESCRIPTION
**Pull request by 2uasimojo**
We recently (#2467) added a tunable poll interval for the clusterversion controller. However, we were being too nice about unparseable values, ignoring them and reverting to the default behavior. With this change, we'll crash the controller instead. The pod will go into Error (or CrashLoopBackoff eventually) with a message telling you what you did wrong.

[HIVE-2619](https://issues.redhat.com//browse/HIVE-2619)

### CODE REVIEW

**Comment by 2uasimojo** [comment]
/override "Red Hat Konflux / hive-operator-enterprise-contract / hive"

File: pkg/controller/clusterversion/clusterversion_controller.go [hunk 1]
```diff
@@ -41,24 +40,33 @@ const (
...
 		logger.WithError(err).Error("could not get controller configurations")
 		return err
 	}
-	return AddToManager(mgr, NewReconciler(mgr, clientRateLimiter), concurrentReconciles, queueRateLimiter)
-}
 
-// NewReconciler returns a new reconcile.Reconciler
-func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) reconcile.Reconciler {
+	var pollInterval time.Duration
+	if envPollInterval := os.Getenv(constants.ClusterVersionPollIntervalEnvVar); len(envPollInterval) > 0 {
```
Comment (suhanime) [inline pkg/controller/clusterversion/clusterversion_controller.go:51]: I missed this earlier but this is great - we should be reading the env earlier when we add to manager so we error out early and not when the reconcile is called

**Review by suhanime — COMMENTED** [review]
/lgtm

**Pull request merged**